include/utils/tree_node.h

```diff
@@ -154,6 +154,15 @@ class TreeNode
                              const std::set<subdomain_id_type> * allowed_subdomains = libmesh_nullptr,
                              Real relative_tol = TOLERANCE) const;
 
+  /**
+   * Get the refinement level of this TreeNode.
+   */
+  unsigned int get_refinement_level() const;
+
+  /**
+   * Set the refinement level of this TreeNode.
+   */
+  void set_refinement_level(unsigned int refinement_level);
 
```

Comment:
Wait a sec, it looks like we already have TreeNode::level that works in the way you propose! Oops. I can update it to use that instead, if you agree.